src/input/process/flow.rs

```diff
@@ -16,139 +18,198 @@ const PROCESS_FLOWS_FILE_NAME: &str = "process_flows.csv";
 struct ProcessFlowRaw {
     process_id: String,
     commodity_id: String,
+    year: String,
+    regions: String,
     flow: f64,
     #[serde(default)]
     flow_type: FlowType,
     flow_cost: Option<f64>,
     is_pac: bool,
 }
 
+impl ProcessFlowRaw {
+    fn validate(&self) -> Result<()> {
+        // Check that flow is not infinity, nan, etc.
+        ensure!(
+            self.flow.is_normal(),
+            "Invalid value for flow ({})",
+            self.flow
+        );
+
+        // **TODO**: https://github.com/EnergySystemsModellingLab/MUSE_2.0/issues/300
+        ensure!(
+            self.flow_type == FlowType::Fixed,
+            "Commodity flexible assets are not currently supported"
+        );
+
+        // Check that flow cost is non-negative
+        if let Some(flow_cost) = self.flow_cost {
+            ensure!(
+                (0.0..f64::INFINITY).contains(&flow_cost),
```

Comment:
Seems like an odd way to write this. Was it a clippy suggestion?

```suggestion
                flow_cost >= 0.0,
```

Reply:
I agree! No idea where this came from - I just moved this code here from somewhere else

Reply:
Actually there's some logic to this - it permits positive values but not infinity

Reply:
Hmm... ok. Let's leave as is